File: db4oj/core/src/com/db4o/internal/ix/IxRemove.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o.internal.ix;

import com.db4o.*;
import com.db4o.foundation.*;
import com.db4o.internal.*;
import com.db4o.internal.freespace.*;

/**
 * A node to represent an entry removed from an Index
 */
public class IxRemove extends IxPatch {

    public IxRemove(IndexTransaction a_ft, int a_parentID, Object a_value) {
        super(a_ft, a_parentID, a_value);
        _size = 0;
    }
    
    public int ownSize() {
        return 0;
    }

    public String toString() {
        if(! Debug4.prettyToStrings){
            return super.toString();
        }
        String str = "IxRemove " + _parentID + "\n " + handler().comparableObject(trans(), _value);
        return str;
    }
    
    public void freespaceVisit(FreespaceVisitor visitor, int index){
        // do nothing
    }
    
    public void visit(Object obj){
        // do nothing
    }
    
    public void visit(Visitor4 visitor, int[] lowerAndUpperMatch){
        // do nothing
    }

    public int write(Indexable4 a_handler, StatefulBuffer a_writer) {
        // do nothing
        return 0;
    }

    public void visitAll(IntObjectVisitor visitor) {
        // do nothing
    }
    
    public Object shallowClone() {
    	IxRemove remove=new IxRemove(_fieldTransaction,_parentID,_value);
    	super.shallowCloneInternal(remove);
    	return remove;
    }
}

File: db4on/core/src/core/com/db4o/internal/ix/IxRemove.cs
namespace com.db4o.@internal.ix
{
	/// <summary>A node to represent an entry removed from an Index</summary>
	public class IxRemove : com.db4o.@internal.ix.IxPatch
	{
		public IxRemove(com.db4o.@internal.ix.IndexTransaction a_ft, int a_parentID, object
			 a_value) : base(a_ft, a_parentID, a_value)
		{
			_size = 0;
		}

		public override int OwnSize()
		{
			return 0;
		}

		public override string ToString()
		{
			return base.ToString();
			string str = "IxRemove " + _parentID + "\n " + Handler().ComparableObject(Trans()
				, _value);
			return str;
		}

		public override void FreespaceVisit(com.db4o.@internal.freespace.FreespaceVisitor
			 visitor, int index)
		{
		}

		public override void Visit(object obj)
		{
		}

		public override void Visit(com.db4o.foundation.Visitor4 visitor, int[] lowerAndUpperMatch
			)
		{
		}

		public override int Write(com.db4o.@internal.ix.Indexable4 a_handler, com.db4o.@internal.StatefulBuffer
			 a_writer)
		{
			return 0;
		}

		public override void VisitAll(com.db4o.foundation.IntObjectVisitor visitor)
		{
		}

		public override object ShallowClone()
		{
			com.db4o.@internal.ix.IxRemove remove = new com.db4o.@internal.ix.IxRemove(_fieldTransaction
				, _parentID, _value);
			base.ShallowCloneInternal(remove);
			return remove;
		}
	}
}
